File: OOP/Dog Shelter App/Project/Exceptions.cpp
#include "Exceptions.h"

ValidationException::ValidationException(std::string message)
{
	this->_message = message;
}

std::string ValidationException::str() const
{
	return this->_message;
}


RepositoryException::RepositoryException(std::string message)
{
	this->_message = message;
}

std::string RepositoryException::str() const
{
	return this->_message;
}

File: Fundamentals of Programming/Activity Planner App/domain/validation/exceptions.py
class ValidationException(Exception):
    def __init__(self, message):
        self._message = message

    def __str__(self):
        return self._message


class RepositoryException(Exception):
    def __init__(self, message):
        self._message = message

    def __str__(self):
        return self._message


class UndoRedoException(Exception):
    def __init__(self, message):
        self._message = message

    def __str__(self):
        return self._message